compiler/testData/codegen/box/ranges/forInReversed/forInReversedArray.kt

```diff
@@ -0,0 +1,14 @@
+// KJS_WITH_FULL_RUNTIME
+// WITH_RUNTIME
+import kotlin.test.*
+
+fun box(): String {
+    val arr = intArrayOf(1, 2, 3, 4)
+    var sum = 0
+    for (i in arr.reversed()) {
```

Comment:
@punzki  reversed/reversedArray() is totatlly unsafe optimization:
```
fun box(): String {
    val arr = intArrayOf(1, 2, 3, 4)
    var sum = 0
    var index = 0
    //for (i in arr.reversedArray()) {
    for (i in arr.reversed()) {
        arr[index++] = 0
        sum = sum * 10 + i
    }
    assertEquals(4321, sum)

    return "OK"
}
```

Reply:
Please add such test cases for reversed and reversedArray and double reversed/reversedArray to don't miss such cases in future

Reply:
@max-kammerer Thank you for pointing this out, I hadn't considered this!
1. Removed optimization for `Array.reversed()/reversedArray()`
2. Added new box tests for what you described
3. Removed bytecode tests for optimized `Array.reversed()/reversedArray()`